### PR TITLE
update boost version requirement to 1.71

### PR DESCRIPTION
**Pull request by greg7mdp**
…ilding on M1 mac

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [x] Documentation Updates
- [ ] Release

### CODE REVIEW

File: Builds/CMake/deps/Boost.cmake [hunk 1]
```diff
@@ -44,7 +44,7 @@ else()
 endif()
 # TBD:
 # Boost_USE_DEBUG_RUNTIME:  When ON, uses Boost libraries linked against the
-find_package(Boost 1.70 REQUIRED
+find_package(Boost 1.75 REQUIRED
```
Comment (seelabs) [inline Builds/CMake/deps/Boost.cmake:47]: 1.71 is the minimum version, I don't think we need to require 1.75.

File: Builds/linux/README.md [hunk 2]
```diff
@@ -33,13 +33,13 @@ protobuf will give errors.
 
 ### Build Boost
 
-Boost 1.70 or later is required. We recommend downloading and compiling boost
+Boost 1.75 or later is required. We recommend downloading and compiling boost
```
Comment (seelabs) [inline Builds/linux/README.md:36]: 1.71 through 1.77. 1.78 and later won't work.

**Review by seelabs — COMMENTED** [review]
Looks good. Left some minor comments about requiring boost 1.75 instead of 1.71.

I'd also do this with two commits instead of one: One commit to update the boost version and one commit for the changes need to compile on arm.

File: Builds/CMake/deps/Boost.cmake (hunk 1 above)
Comment (greg7mdp) [reply Builds/CMake/deps/Boost.cmake:47]: Scott advised to use 1.75 and that's what I used on both linux and macos. I feel better recommending something I have tried, but I can change to 1.71 if you prefer.

File: Builds/VisualStudio2017/README.md [hunk 3]
```diff
@@ -80,13 +80,13 @@ to get the correct 32-/64-bit variant.
...
+Boost 1.75 or later is required.
 
 After [downloading boost](http://www.boost.org/users/download/) and unpacking it
-to `c:\lib`. As of this writing, the most recent version of boost is 1.70.0,
-which will unpack into a directory named `boost_1_70_0`. We recommended either
+to `c:\lib`. As of this writing, the recommended version of boost is 1.75.0,
+which will unpack into a directory named `boost_1_75_0`. We recommended either
 renaming this directory to `boost`, or creating a junction link `mklink /J boost
-boost_1_70_0`, so that you can more easily switch between versions.
+boost_1_75_0`, so that you can more easily switch between versions.
```
[the remark this answers is not in the document]
Comment (greg7mdp) [reply Builds/VisualStudio2017/README.md:89]: Thanks Scott, will split into two commits, and add the note about 1.78+

**Review by seelabs — APPROVED** [review]
:+1: LGTM

File: Builds/linux/README.md [hunk 4]
```diff
@@ -88,14 +88,14 @@ git checkout develop
...
+For example, you built Boost in your home directory `~/boost_1_71_0`, you
 would do for any shell in which you want to build:
 
 ```
-export BOOST_ROOT=~/boost_1_70_0
+export BOOST_ROOT=~/boost_1_71_0
 ```
 
-Alternatively, you can add `DBOOST_ROOT=~/boost_1_70_0` to the command line when
+Alternatively, you can add `DBOOST_ROOT=~/boost_1_71_0` to the command line when
```
Comment (legleux) [inline Builds/linux/README.md:98]: While this file is being touched, this parameter is missing the `-` i.e. `-DBOOST_ROOT=~/boost_1_71_0`
I like copy/pasteable things.

**Review by legleux — CHANGES_REQUESTED** [review]

File: Builds/linux/README.md (hunk 4 above)
Comment (greg7mdp) [reply Builds/linux/README.md]: done, thanks for the suggestion @legleux !

**Review by legleux — APPROVED** [review]

**Pull request closed without merging**